src/test/java/com/google/cloud/tools/gradle/appengine/core/DeployAllTaskTest.java

```diff
@@ -92,26 +93,85 @@ public void testDeployAllAction() throws AppEngineException, IOException {
   }
 
   @Test
-  public void testDeployAllAction_validFileNotInDir() throws AppEngineException, IOException {
+  public void testDeployAllAction_flexible() throws AppEngineException, IOException {
+    deployConfig.setAppEngineDirectory(tempFolder.newFolder("appengine"));
+
+    final File appYaml = tempFolder.newFile("staging/app.yaml");
+    final File cronYaml = tempFolder.newFile("appengine/cron.yaml");
+    final File dispatchYaml = tempFolder.newFile("appengine/dispatch.yaml");
+    final File dosYaml = tempFolder.newFile("appengine/dos.yaml");
+    final File indexYaml = tempFolder.newFile("appengine/index.yaml");
+    final File queueYaml = tempFolder.newFile("appengine/queue.yaml");
+    final File invalidYaml = tempFolder.newFile("appengine/invalid.yaml");
+
+    deployAllTask.deployAllAction();
+
+    verify(deploy).deploy(deployCapture.capture());
+    DeployConfiguration captured = deployCapture.getValue();
+    assertTrue(captured.getDeployables().contains(appYaml));
+    assertTrue(captured.getDeployables().contains(cronYaml));
+    assertTrue(captured.getDeployables().contains(dispatchYaml));
+    assertTrue(captured.getDeployables().contains(dosYaml));
+    assertTrue(captured.getDeployables().contains(indexYaml));
+    assertTrue(captured.getDeployables().contains(queueYaml));
+    assertFalse(captured.getDeployables().contains(invalidYaml));
+  }
+
+  @Test
+  public void testDeployAllAction_validFileNotInDirStandard()
+      throws AppEngineException, IOException {
+    deployConfig.setAppEngineDirectory(stageDir);
+
+    final File appYaml = tempFolder.newFile("staging/app.yaml");
+    final File validInDifferentDirYaml = tempFolder.newFile("queue.yaml");
+
+    deployAllTask.deployAllAction();
+
+    verify(deploy).deploy(deployCapture.capture());
+    DeployConfiguration captured = deployCapture.getValue();
+    assertTrue(captured.getDeployables().contains(appYaml));
+    assertFalse(captured.getDeployables().contains(validInDifferentDirYaml));
+  }
+
+  @Test
+  public void testDeployAllAction_validFileNotInDirFlexible()
+      throws AppEngineException, IOException {
+    deployConfig.setAppEngineDirectory(tempFolder.newFolder("appengine"));
+
     // Make YAMLS
-    File appYaml = tempFolder.newFile("staging/app.yaml");
-    File validInDifferentDirYaml = tempFolder.newFile("queue.yaml");
+    final File appYaml = tempFolder.newFile("staging/app.yaml");
+    final File validInDifferentDirYaml = tempFolder.newFile("queue.yaml");
+
+    deployAllTask.deployAllAction();
+
+    verify(deploy).deploy(deployCapture.capture());
+    DeployConfiguration captured = deployCapture.getValue();
+    assertTrue(captured.getDeployables().contains(appYaml));
+    assertFalse(captured.getDeployables().contains(validInDifferentDirYaml));
+  }
+
+  @Test
+  public void testDeployAllAction_configNotModified() throws AppEngineException, IOException {
+    deployConfig.setAppEngineDirectory(stageDir);
```

Comment:
I think this test might cover more bases if it put something in the deployables for the original config.